drivers/volume/linstor/linstor.go

```diff
@@ -0,0 +1,412 @@
+		if err == nil {
+			if rack, ok := labels[rackLabelKey]; ok {
+				newInfo.Rack = rack
+			}
+			if zone, ok := labels[v1.LabelZoneFailureDomain]; ok {
+				newInfo.Zone = zone
+			}
+			if region, ok := labels[v1.LabelZoneRegion]; ok {
+				newInfo.Region = region
+			}
+		}
+		infos = append(infos, newInfo)
+	}
+	return infos, nil
+}
+
+func (l *linstor) GetPodVolumes(podSpec *v1.PodSpec, namespace string) ([]*storkvolume.Info, error) {
+	logrus.Infof("called: GetPodVolumes(%s, %s)", podSpec.Containers[0].Name, namespace)
+	var volumes []*storkvolume.Info
+	for _, volume := range podSpec.Volumes {
+		volumeName := ""
+		if volume.PersistentVolumeClaim != nil {
+			pvc, err := core.Instance().GetPersistentVolumeClaim(
+				volume.PersistentVolumeClaim.ClaimName,
+				namespace)
+			if err != nil {
+				return nil, err
+			}
+
+			if !l.OwnsPVC(pvc) {
+				continue
+			}
+
+			if pvc.Status.Phase == v1.ClaimPending {
+				return nil, &storkvolume.ErrPVCPending{
+					Name: volume.PersistentVolumeClaim.ClaimName,
+				}
+			}
+			volumeName = pvc.Spec.VolumeName
+		}
+
+		if volumeName != "" {
+			volumeInfo, err := l.InspectVolume(volumeName)
+			if err != nil {
+				// If the ispect volume fails return with atleast some info
+				volumeInfo = &storkvolume.Info{
+					VolumeName: volumeName,
+				}
+			}
+			volumes = append(volumes, volumeInfo)
+		}
+	}
+	return volumes, nil
+}
+
+func (l *linstor) GetVolumeClaimTemplates(templates []v1.PersistentVolumeClaim) ([]v1.PersistentVolumeClaim, error) {
+	logrus.Infof("called: GetVolumeClaimTemplates([%d pvcs])", len(templates))
+	var linstorTemplates []v1.PersistentVolumeClaim
+	for _, t := range templates {
+		if l.OwnsPVC(&t) {
+			linstorTemplates = append(linstorTemplates, t)
+		}
+	}
+	return linstorTemplates, nil
+}
+
+func (l *linstor) OwnsPVC(pvc *v1.PersistentVolumeClaim) bool {
+	logrus.Infof("called: OwnsPVC(%s)", pvc.ObjectMeta.Name)
+
+	provisioner := ""
+	// Check for the provisioner in the PVC annotation. If not populated
+	// try getting the provisioner from the Storage class.
+	if val, ok := pvc.Annotations[pvcProvisionerAnnotation]; ok {
+		provisioner = val
+	} else {
+		storageClassName := k8shelper.GetPersistentVolumeClaimClass(pvc)
+		if storageClassName != "" {
+			storageClass, err := storage.Instance().GetStorageClass(storageClassName)
+			if err == nil {
+				provisioner = storageClass.Provisioner
+			} else {
+				logrus.Warnf("Error getting storageclass %v for pvc %v: %v", storageClassName, pvc.Name, err)
+			}
+		}
+	}
+
+	if provisioner == "" {
+		// Try to get info from the PV since storage class could be deleted
+		pv, err := core.Instance().GetPersistentVolume(pvc.Spec.VolumeName)
+		if err != nil {
+			logrus.Warnf("Error getting pv %v for pvc %v: %v", pvc.Spec.VolumeName, pvc.Name, err)
+			return false
+		}
+		logrus.Infof("No information in PVC, try PV?")
+		return l.OwnsPV(pv)
+	}
+
+	if provisioner == provisionerName {
+		logrus.Infof("-> yes")
```

Comment:
With multiple requests this log will be interleaved with other messages and won't provide any info